edx_exams/apps/lti/tests/test_views.py

```diff
@@ -1,20 +1,248 @@
 """
 Tests for the exams LTI views
 """
+import json
 import uuid
 from unittest.mock import patch
 from urllib.parse import urljoin
 
+import ddt
+from Cryptodome.PublicKey import RSA
+from django.contrib.auth.models import AnonymousUser
 from django.urls import reverse
 from lti_consumer.data import Lti1p3LaunchData, Lti1p3ProctoringLaunchData
-from lti_consumer.models import LtiConfiguration
+from lti_consumer.lti_1p3.extensions.rest_framework.authentication import Lti1p3ApiAuthentication
+from lti_consumer.models import LtiConfiguration, LtiProctoringConsumer
 
 from edx_exams.apps.api.test_utils import ExamsAPITestCase, UserFactory
 from edx_exams.apps.core.models import CourseExamConfiguration, Exam, ExamAttempt
 from edx_exams.apps.core.statuses import ExamAttemptStatus
 from edx_exams.apps.lti.utils import get_lti_root
 
 
+@ddt.ddt
+class LtiAcsTestCase(ExamsAPITestCase):
```

Comment:
we could probably use some limited testing on auth here. It's mostly handled in the library but it's common to have something basic to make sure everything configured properly. For example when we do: https://github.com/edx/edx-exams/blob/e3e375993afe46b0bee6db91b163f5f0ee10a69b/edx_exams/apps/api/v1/tests/test_views.py#L82

Maybe in this case we just check for no token and a token without the correct scope.

Reply:
I just pushed some code with modifications regarding all of the other comments. I also created a test for when authentication fails.

I tried making a test where the token is valid but doesn't have the correct scope and had some issues (I can go into more detail about this if you'd like). I noticed that we already test the permissions class for the correct scope within xblock-lti-consumer ([Link](https://github.com/openedx/xblock-lti-consumer/blob/master/lti_consumer/lti_1p3/tests/extensions/rest_framework/test_permissions.py#L281)), so I'm wondering if there's a reason we should still test for the scope within this repo.

Reply:
ahh ok usually it's pretty easy to test so we do but I see how you'd have to setup an entire config with a non-proctoring consumer which is weird so let's leave this one out.